src/block-management/ethrpc-transport-adapter.js

```diff
@@ -0,0 +1,41 @@
+"use strict";
+
+var nextToken = 1;
+var subscriptionMapping = {};
+
+module.exports = function (ethrpc) {
+  return {
+    getBlockByNumber: ethrpc.getBlockByNumber.bind(ethrpc),
+    getBlockByHash: ethrpc.getBlockByHash.bind(ethrpc),
+    subscribeToReconnects: ethrpc.internalState.transporter.addReconnectListener.bind(ethrpc.internalState.transporter),
+    unsubscribeFromReconnects: ethrpc.internalState.transporter.removeReconnectListener.bind(ethrpc.internalState.transporter),
+    subscribeToNewHeads: function (onNewHead, onSubscriptionError) {
+      var token = (nextToken++).toString();
+      subscriptionMapping[token] = null;
+      try {
```

Comment:
What errors can be picked up by this try/catch?  Since all that's inside the try is an async function and its callback, would it make more sense to put the try/catch inside the callback (possibly w/ catch passing the exception to `onSubscriptionError`)?